File: app/src/main/java/com/benhero/glstudio/base/GLTranslateAnimation.java
package com.benhero.glstudio.base;

import android.opengl.Matrix;

/**
 * GL平移动画
 *
 * @author chenbenbin
 */
public class GLTranslateAnimation extends GLAnimation {
    private float mFromX;
    private float mFromY;
    private float mToX;
    private float mToY;

    public float getFromX() {
        return mFromX;
    }

    public void setFromX(float fromX) {
        mFromX = fromX;
    }

    public float getFromY() {
        return mFromY;
    }

    public void setFromY(float fromY) {
        mFromY = fromY;
    }

    public float getToX() {
        return mToX;
    }

    public void setToX(float toX) {
        mToX = toX;
    }

    public float getToY() {
        return mToY;
    }

    public void setToY(float toY) {
        mToY = toY;
    }

    @Override
    protected void applyTransformation(float percent, GLObject object) {
        super.applyTransformation(percent, object);
        float currentX = (mToX - mFromX) * percent + mFromX;
        float currentY = (mToY - mFromY) * percent + mFromY;
        Matrix.translateM(object.getPositionMatrix(), 0,
                object.xPositionToGL(currentX), object.yPositionToGL(currentY), 0);
    }
}

File: app/src/main/java/com/benhero/glstudio/base/GLTranslateAnimation.kt
package com.benhero.glstudio.base

import android.opengl.Matrix

/**
 * GL平移动画
 *
 * @author chenbenbin
 */
class GLTranslateAnimation : GLAnimation() {
    var fromX: Float = 0.toFloat()
    var fromY: Float = 0.toFloat()
    var toX: Float = 0.toFloat()
    var toY: Float = 0.toFloat()

    override fun applyTransformation(percent: Float, glObject: GLObject) {
        super.applyTransformation(percent, glObject)
        val currentX = (toX - fromX) * percent + fromX
        val currentY = (toY - fromY) * percent + fromY
        Matrix.translateM(glObject.positionMatrix, 0,
                glObject.xPositionToGL(currentX), glObject.yPositionToGL(currentY), 0f)
    }
}
